lib/db.c

```diff
@@ -1598,10 +1598,12 @@ grn_table_create_validate(grn_ctx *ctx,
     return ctx->rc;
   }
 
-  if ((flags & GRN_OBJ_KEY_LARGE) && table_type != GRN_OBJ_TABLE_HASH_KEY) {
+  if ((flags & GRN_OBJ_KEY_LARGE) && (table_type != GRN_OBJ_TABLE_HASH_KEY &&
+                                      table_type != GRN_OBJ_TABLE_PAT_KEY)) {
     ERR(GRN_INVALID_ARGUMENT,
         "[table][create] "
-        "large key support is available only for TABLE_HASH_KEY key table: "
+        "large key support is available for TABLE_HASH_KEY and TABLE_PAT_KEY "
+        "key table: "
         "<%.*s>(%s)",
```

Comment:
```suggestion
        "large key support is available for TABLE_HASH_KEY and TABLE_PAT_KEY "
        "key table: <%.*s>(%s)",
```

No need for a line break?

Reply:
Ah... This is correct.
Thank you for your comment.